Index: libproxy/cmake/modules/pacrunner_webkit.cmk
===================================================================
--- libproxy/cmake/modules/pacrunner_webkit.cmk	(revision 815)
+++ libproxy/cmake/modules/pacrunner_webkit.cmk	(working copy)
@@ -13,7 +13,10 @@
   endif()
 else()
   if(WITH_WEBKIT3)
-    px_check_modules(WEBKIT webkitgtk-3.0)
+    px_check_modules(WEBKIT javascriptcoregtk-3.0>=1.5.0)
+    if(NOT WEBKIT_LIBRARIES)
+      px_check_modules(WEBKIT webkitgtk-3.0<1.5.0)
+    endif(NOT WEBKIT_LIBRARIES)
   else()
     px_check_modules(WEBKIT webkit-1.0)
   endif()

